app/components/profile_picture_picker/index.tsx

```diff
@@ -0,0 +1,454 @@
+                case Permissions.RESULTS.DENIED:
+                    permissionRequest = await Permissions.request(
+                        storagePermission,
+                    );
+                    if (permissionRequest !== Permissions.RESULTS.GRANTED) {
+                        return false;
+                    }
+                    break;
+                case Permissions.RESULTS.BLOCKED: {
+                    const {title, text} = getPermissionDeniedMessage(storagePermission);
+
+                    Alert.alert(title, text, [
+                        {
+                            text: formatMessage({
+                                id: 'mobile.permission_denied_dismiss',
+                                defaultMessage: "Don't Allow",
+                            }),
+                        },
+                        {
+                            text: formatMessage({
+                                id: 'mobile.permission_denied_retry',
+                                defaultMessage: 'Settings',
+                            }),
+                            onPress: () => AndroidOpenSettings.appDetailsSettings(),
+                        },
+                    ]);
+                    return false;
+                }
+            }
+        }
+
+        return true;
+    };
+
+    const prepareFileUpload = async (files: File[]) => {
+        const file = files?.[0];
+
+        if (!file) {
+            return;
+        }
+
+        if (!file.fileSize || !file.fileName) {
+            const path = Platform.OS === 'ios' ? (file?.path || file.uri).replace('file://', '') : (file?.path || file.uri);
+            const fileInfo = await FileSystem.getInfoAsync(path);
+            const uri = fileInfo.uri;
+            file.fileSize = fileInfo.size;
+            file.fileName = uri.substr(uri.lastIndexOf('/') + 1);
+        }
+
+        if (!file.type) {
+            file.type = lookupMimeType(file.fileName) as typeof VALID_MIME_TYPES[number];
+        }
+
+        if (!Files.VALID_MIME_TYPES.includes(file.type)) {
+            onShowUnsupportedMimeTypeWarning();
+        } else if (file!.fileSize! > maxFileSize) {
+            onShowFileSizeWarning(file.fileName!);
+        } else {
+            DeviceEventEmitter.emit(Navigation.NAVIGATION_CLOSE_MODAL);
+            uploadFiles(files);
+        }
+    };
+
+    const getRemoveProfileImageOption = () => {
+        const {id, lastPictureUpdate} = currentUser;
+
+        let client: Client | undefined;
+        let profileImageUrl: string | undefined;
+
+        try {
+            client = NetworkManager.getClient(serverUrl);
+            profileImageUrl = client.getProfilePictureUrl(id, lastPictureUpdate);
+        } catch {
+            // does nothing
+        }
+
+        // Check if image url includes query string for timestamp. If so, it means the image has been updated from the default, i.e. '.../image?_=1544159746868'
+        if (profileImageUrl?.includes('?')) {
```

Comment:
We are verifying the result from `getProfilePictureUrl` method of the rest client.  If it does contain `?`, then it would mean that the user has an existing profile image on the server.  Am I missing anything?

Reply:
I am trying to think of any other weird situation where a `?` may appear in that url (like a `?` in the base url? (that should not be possible as far as I know)). Maybe I am worrying too much and this is fine as it is.

Reply:
I can check for `image?_` instead of only `?`